### DISCUSSION TITLE
[Bug]: Several TypeScript errors

### DISCUSSION THREAD

**Discussion by peakercope** [react-hook-form/react-hook-form#6132] [Q&A]
### Version Number

7.12.1

### Codesandbox

https://codesandbox.io/s/quizzical-black-qem07?file=/src/App.tsx

### Steps to reproduce

Please look at [https://codesandbox.io/s/quizzical-black-qem07?file=/src/App.tsx](https://codesandbox.io/s/quizzical-black-qem07?file=/src/App.tsx)

I'm receiving this typescript errors when using `useFieldArray` hook.

### Expected behaviour

No TypeScript error messages.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
use ` as const` , it's limitation.

**Reply by peakercope** [reply]
@bluebill1049 thanks for quick answer!
there is another error with `setValue` could you please look at
![image](https://user-images.githubusercontent.com/10975462/127733247-fb8e8bf4-483b-4cba-8fbf-733e440b3e0f.png)

[csb](https://codesandbox.io/s/quizzical-black-qem07?file=/src/App.tsx)

**Reply by bluebill1049** [reply]
```setValue(`list.${index}.listName` as `list.0.listName`, value);```

**Reply by peakercope** [reply]
@bluebill1049  one more question🙏
as value, returned from `getValues` typed as `never` I can't apply array methods on it
https://codesandbox.io/s/quizzical-black-qem07?file=/src/App.tsx
